.github/workflows/release.yml

```diff
@@ -0,0 +1,217 @@
+    # superfluous.
+    - name: build mfem
+      if: steps.cache-mfem.outputs.cache-hit != 'true'
+      uses: mfem/github-actions/build-mfem@v2.2
+      with:
+        os: ${{ matrix.os }}
+        target: ${{ matrix.target }}
+        hypre-dir: ${{ env.HYPRE_TOP_DIR }}
+        metis-dir: ${{ env.METIS_TOP_DIR }}
+        mfem-dir: ${{ env.MFEM_TOP_DIR }}
+        mfem-branch: ${{ env.MFEM_BRANCH }}
+        build-system: ${{ matrix.build-system }}
+        mpi: ${{ matrix.mpi }}
+
+    # Install GLVis dependencies with package manager
+    - name: get deps (Linux)
+      if: matrix.os == 'ubuntu-20.04'
+      run: |
+        # We need to add a PPA for SDL 2.0.14 - fixes some initialization
+        # errors for X11
+        sudo add-apt-repository -y ppa:savoury1/multimedia
+        sudo apt-get update
+        sudo apt-get install libfontconfig1-dev libfreetype6-dev libsdl2-dev libglew-dev libglm-dev libpng-dev
+
+    - name: Set up Homebrew
+      if: matrix.os == 'macos-10.15'
+      uses: Homebrew/actions/setup-homebrew@master
+
+    - name: get deps (MacOS)
+      if: matrix.os == 'macos-10.15'
+      run: |
+        export HOMEBREW_NO_INSTALL_CLEANUP=1
+        brew install fontconfig freetype sdl2 glew glm libpng
+
+    - name: cache deps (Windows)
+      id: cache-deps
+      uses: actions/cache@v3
+      with:
+        path: vcpkg_cache
+        key: ${{ runner.os }}-vcpkg-v1
+
+    - name: prepare binary cache location
+      if: steps.cache-deps.outputs.cache-hit != 'true'
+      run: |
+        mkdir -p vcpkg_cache
+
+    - name: checkout GLVis
+      uses: actions/checkout@v2
+      with:
+        path: glvis
+        submodules: recursive
+
+    - name: build GLVis (cmake)
+      if: matrix.build-system == 'cmake'
+      env:
+        VCPKG_DEFAULT_BINARY_CACHE: ${{ github.workspace }}/vcpkg_cache
+      run: |
+        build_type="Release"
+        [[ ${{ matrix.os }} == "windows-2022" ]] \
+          && toolchain_file="${VCPKG_INSTALLATION_ROOT}\\scripts\\buildsystems\\vcpkg.cmake"
+        cd glvis && mkdir build && cd build
+        cmake \
+          -D CMAKE_TOOLCHAIN_FILE:STRING=${toolchain_file} \
+          -D CMAKE_BUILD_TYPE:STRING=${build_type} \
+          -D ENABLE_TESTS:BOOL=TRUE \
+          -D mfem_DIR:PATH=${GITHUB_WORKSPACE}/${MFEM_TOP_DIR}/build \
+          -D GLVIS_BASELINE_SYS=${{ matrix.os }} \
+          ..
+        cmake --build . --parallel 3 --config "${build_type}"
+      shell: bash
+
+    - name: package binary (Windows)
+      if: matrix.os == 'windows-2022'
+      env:
+        GLVIS_EXPORT_NAME: glvis-${{ github.ref_name }}-${{ runner.os }}-amd64
+      run: |
+        cd glvis/build
+        Copy-Item -Path "Release" -Destination "${Env:GLVIS_EXPORT_NAME}" -Recurse
+
+    - name: package binary (Mac)
+      if: matrix.os == 'macos-10.15'
+      env:
+        GLVIS_EXPORT_NAME: glvis-${{ github.ref_name }}-${{ runner.os }}-amd64
```

Comment:
After

```
chmod +x GLVis.app/Contents/MacOS/glvis 
```

and right-click/Open to get past Gatekeeper it works on Catalina.

But not on BigSur. I get this error there:

<img width="372" alt="Screen Shot 2022-05-18 at 3 44 03 PM" src="https://user-images.githubusercontent.com/1278247/169167451-c09ee30e-1ba5-4fce-b650-a0908164c763.png">

Reply:
Is your Mac on Big Sur an M1 Mac? I would imagine there would be backwards compatibility issues if we were building on Big Sur and running on Catalina, but not when it's the other way around.

Reply:
Nope, it is Intel.

Here's what I get when I run it from the command line:

```
> GLVis.app/Contents/MacOS/glvis

dyld: Library not loaded: /usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib
  Referenced from: GLVis.app/Contents/MacOS/glvis
  Reason: image not found
Abort
```

Is there an expectation that SDL2 is globally installed? (that's not the case for me)

Reply:
So there were two issues:
- We needed to copy the dylibs used by glvis into a `Frameworks` directory, then use `install_name_tool` to point the glvis executable to the bundle-local dylibs. This is achieved with the install-time `fixup_bundle` function in CMake.
- `actions/upload-artifact` apparently clobbers our Unix execute bit on the app bundle. To avoid this, we create a .dmg to hold our `GLVis.app` file.

Can you see if the new release build works for you?

Reply:
This works great, and in addition the binary is smaller 😄 

Thanks!